javaee-like-getting-started/src/test/java/io/quarkus/ts/javaee/gettingstarted/OpenAPITracingIT.java

```diff
@@ -0,0 +1,130 @@
+    private static final String OPEN_API_PATH = "/q/openapi";
+    private static final String OPEN_API_OPERATION = "GET /openapi";
+    private static final String SWAGGER_UI_PATH = "/q/swagger-ui/";
+    private static final String SWAGGER_UI_OPERATION = "GET /swagger-ui";
+
+    @JaegerContainer(expectedLog = "\"Health Check state change\",\"status\":\"ready\"")
+    static final JaegerService jaeger = new JaegerService();
+
+    @QuarkusApplication(classes = GreetingResource.class)
+    static final RestService tracedApp = new RestService()
+            .withProperty("quarkus.application.name", "tracedApp")
+            .withProperty("quarkus.otel.exporter.otlp.traces.endpoint", jaeger::getCollectorUrl)
+            .withProperty(SUPPRESS_NON_APPLICATION_URIS, "false");
+
+    @QuarkusApplication(classes = GreetingResource.class)
+    static final RestService untracedApp = new RestService()
+            .withProperty("quarkus.application.name", "untracedApp")
+            .withProperty("quarkus.otel.exporter.otlp.traces.endpoint", jaeger::getCollectorUrl)
+            .withProperty(SUPPRESS_NON_APPLICATION_URIS, "true");
+
+    @Test
+    public void openApiTraced() {
+        testTraced(OPEN_API_PATH, OPEN_API_OPERATION);
+    }
+
+    @Test
+    public void swaggerUiTraced() {
+        testTraced(SWAGGER_UI_PATH, SWAGGER_UI_OPERATION);
+    }
+
+    @Test
+    public void openApiUntraced() {
+        testUntraced(OPEN_API_PATH, OPEN_API_OPERATION);
+    }
+
+    @Test
+    @Disabled("https://github.com/quarkusio/quarkus/issues/34376")
+    public void swaggerUiUntraced() {
+        testUntraced(SWAGGER_UI_PATH, SWAGGER_UI_OPERATION);
+    }
+
+    private void testTraced(String path, String operation) {
+        callEndpoint(tracedApp, path);
+        await()
+                .atMost(10, TimeUnit.SECONDS)
+                .pollInterval(1, TimeUnit.SECONDS)
+                .untilAsserted(() -> {
+                    Response response = retrieveTraces(tracedApp.getName(), operation);
+                    verifyDataSize(response, 1);
+                    verifySpansSize(response);
+                    verifySpanOperationName(response, operation);
+                    verifySpanTag(response, HTTP_TARGET_TAG, path);
+                    verifySpanTag(response, HTTP_STATUS_CODE_TAG, HttpStatus.SC_OK);
+                    verifySpanTag(response, HTTP_METHOD_TAG, "GET");
+                    verifySpanTag(response, SPAN_KIND_TAG, "server");
+                });
+    }
+
+    private void testUntraced(String path, String operation) {
+        callEndpoint(untracedApp, path);
+        await()
+                .during(10, TimeUnit.SECONDS)
+                .atMost(12, TimeUnit.SECONDS)
+                .pollInterval(1, TimeUnit.SECONDS)
+                .untilAsserted(() -> {
+                    Response response = retrieveTraces(untracedApp.getName(), operation);
+                    verifyDataSize(response, 0);
```

Comment:
strictly speaking, this condition should be reversed, right? You wait until you receive 0, but you can also receive 0 before traces arrives (that depends on exported when it sends traces and how quickly they get processed so that they are available), it is possible results will differ in CI and your laptop as locally traces may arrive immediately (fast) but in CI it won't.

I don't suggest to change it though :-/ we would waste time to something that is not coming, just want to check you know about this? Really don't know...

Reply:
RN, it waits for at most 12 seconds, during which the value must be 0 for at least 10 seconds.

What do you mean by reverse?

Reply:
I do read `org.awaitility.core.ConditionFactory#ConditionFactory` differently than you do (`atMost` deletes `during`). However if you are sure you can apply 2 `org.awaitility.constraint.WaitConstraint`, I believe you and I read it wrong. I'll test it when I find a time and resolve comment. Thank you.

Reply:
Yeah, I think they are chaining it.